battle-engine.js

```diff
@@ -1633,6 +1633,14 @@ Battle = (function () {
 		if (source === undefined && this.event && this.event.target) source = this.event.target;
 
 		if (this.weather === status.id && this.gen > 2) return false;
+		if (status.id && !this.runEvent('SetWeather', source, source, status)) {
```

Comment:
You should possibly record the result of `runEvent` so you can display the error message depending on whether `null` or `false` is returned.